client/add/add.ctrl.js

```diff
@@ -123,11 +128,16 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
-      .tap(box.onscroll)
+      .filter(line => (boxCounts[line.box]++ % BOX_PAGE_SIZE) < BOX_PAGE_SIZE)
```

Comment:
Now the function will always return true, because `anything % x < x`.

Just to make sure, have I explained the intended behavior clearly enough? I feel like we're miscommunicating about what this is supposed to do.